src/libraries/Native/Unix/System.Native/pal_networking.c

```diff
@@ -3085,6 +3085,11 @@ int32_t SystemNative_Disconnect(intptr_t socket)
     addr.sa_family = AF_UNSPEC;
 
     err = connect(fd, &addr, sizeof(addr));
+    if (err != 0) 
+    {
+        // On some older kernels connect(AF_UNSPEC) may fail. Fall back to shutdown in these cases:
```

Comment:
`connect(AF_UNSPEC)` has worked since old kernels. It's an issue in SELinux that caused this to fail. That's why it is affecting SELinux based distros. You don't need to update the comment, this is background info.